### DISCUSSION TITLE
grain boundary

### DISCUSSION THREAD

**Discussion by mugikumar** [mtex-toolbox/mtex#1807] [Ask Anything]
Hi developers,
                I am working on grain boundary plane distribution. From the misorientation, angle distribution found a certain angle and axis. how to define the certain misorientation angle in condition Mtex code
```matlab
cond = gB.misorientation.angle > 57 * degree
contourf(gbnd1,'colorrange',[0.8 1.5])
mtexTitle('GBPD for misorientation angle \(> 57^{\circ}\)')
mtexColorMap parula
```
whenever I load this code its gives in the stereographic projection, not in triangular like Mtex docuentation

**Comment by kilir** [comment]
Hi,
what is the output if you type `gbnd1` in the command window? S2FunHarmonicSym? 

If it is, mtex should actully resort to just plotting the fundamental sector.  If it is not, it might help if you'd post the relevant lines where you compute `gbnd1`.

Cheers,
Rüdiger

**Reply by mugikumar** [reply]
Hi kilir,
      Thanks for your reply, yes its showing S2FunHarmonicSym.
![mtex contour](https://user-images.githubusercontent.com/109027461/235659090-3db11f5b-d201-4aa9-84bb-1c842324f344.jpg)
i have to plot the traces of the grain boundaries also.

**Reply by kilir** [reply]
Maybe in your case you have to add `'fundamentalSector'` as an argument to the plotting function.
Cheers,
Rüdiger

**Reply by mugikumar** [reply]
Hi Kilir'
Sorry, it's not working

**Reply by mugikumar** [reply]
Hi kilir,
      Sorry to bother you again. I have a another doubt 
rot = rotation.byAxisAngle(xvector,90*degree);
if i want to rotate by 110 axis what is the code could you please help me?

**Reply by kilir** [reply]
What does `gbnd1.CS` return in your case?

**Reply by mugikumar** [reply]
Hi kilir
it showed triclinic specimen symmetry

**Reply by kilir** [reply]
That should be a crystal symmetry. Please post the relevant code you are using.

**Reply by mugikumar** [reply]
```matlab
gB = [grains.boundary('indexed','indexed') grains('indexed','indexed').innerBoundary]
cond = gB.misorientation.angle > 57 * degree
gbnd1 = calcGBPD(gB(cond),ebsd,'fundamentalSector','halfwidth',5*degree)

contourf(gbnd1,'fundamentalSector','colorrange',[0.8 1.5])
mtexTitle('gbpd')
mtexColorMap parula
```

**Reply by kilir** [reply]
And what is the output of `grains`? 
And are you sure you only have one indexed phase?

**Comment by mugikumar** [comment]
Yes it is bcc martensite phase only and grains output is nearly 650*1

**Reply by kilir** [reply]
Which also loads as a cubic phase I presume? The output of `grains` should not just be a number but also tell you about the phases and their symmetry.

**Reply by mugikumar** [reply]
Hi kilir,
           Sorry for my previous reply, yes its cubic phase and the output of grains also iron bcc (m-3m)
 
![mtex doubt](https://user-images.githubusercontent.com/109027461/235989810-3b976f1c-6dac-4c00-abd6-bde0d16acbf2.jpg)

**Reply by kilir** [reply]
Can you provide a sample of your data where one can reproduce your issue?

**Reply by mugikumar** [reply]
Hi Kilir,
          sorry for late reply, how to provide my data on here.

**Reply by kilir** [reply]
Either use a small map where you can reproduce the issue, or take a subset of a larger map once loaded into mtex, save as .mat, zip it and put it here.

**Reply by mugikumar** [reply]
Hi kilir,
          I have uploaded the EBSD document of the workspace in mat files.
[10martensite.zip](https://github.com/mtex-toolbox/mtex/files/11418016/10martensite.zip)

**Reply by kilir** [reply]
Hi,
running your datra through the documentation `BoundaryNormalDistribution.m` I get the following:

<img width="976" alt="mugiku" src="https://user-images.githubusercontent.com/11697557/236756725-7bebc5c0-a858-4b4d-b2a3-0c74bea6508f.png">

I assume you use the latest mtex version and a recent matlab, right?

Cheers,
Rüdiger

**Reply by mugikumar** [reply]
Hi kilir,
         Thanks for your continuous helping me. Yes, I am using the Matlab of 2022b and Mtex of 5.9.0. Anyway, I will check for any update for Matlab and rectify it if there is. Is it possible to rotate in a specific [uvw] direction and misorientation angle in the Mtex code?

**Reply by kilir** [reply]
And when you run exactly the `BoundaryNormalDistribution.m` with our input data, you do not get a  `' S2FunHarmonicSym '` as an output?

**Comment by mugikumar** [comment]
No i am getting s2 fun harmonic symmetry when i run of GBPD mtex code

**Reply by kilir** [reply]
Great, then everything is fine, right?

Cheers,
Rüdiger

**Comment by mugikumar** [comment]
Hi kilir,
         No it was plotting in stereographic mode not in contourf when i
was running the code but i dont know how you are getting could you please
send the code?

**Reply by kilir** [reply]
These are different things. If you mean with "stereographic mode" a type of spherical projection, whihc is an equal angle projection. The default in Mtex is an equal area projection. Whether you have filled contours (`'contourf'`) just contour lines (`'contour'`) or a smooth plot is just a plotting option.

If you plot a spherical function of type `S2FunHarmonicSym`, by default the dispalyed part will be the fundamental sector according to the symmetry of the function.

I used the code as it is in the documentation, no modifications.

Cheers,
Rüdiger

**Reply by mugikumar** [reply]
Thanks kilir
                  I got it. Is this possible to plot a certain misorientation angle with rotation at a specific [uvw]? Because now, we are considering the grain boundary misorientation of all directions and then rotating it like the plot in TSL OIM.
